unittests/AST/ASTImporterTest.cpp

```diff
@@ -2689,6 +2689,157 @@ TEST_P(ASTImporterTestBase, ImportDefinitionOfClassTemplateAfterFwdDecl) {
   }
 }
 
+TEST_P(ASTImporterTestBase, ImportOfEquivalentRecordWithField) {
+  Decl *ToD1;
+  {
+    Decl *FromTU = getTuDecl(
+        R"(
+            struct A { int x; };
+            )",
+        Lang_CXX, "input0.cc");
+    auto *FromD = FirstDeclMatcher<CXXRecordDecl>().match(
+        FromTU, cxxRecordDecl(hasName("A")));
+
+    ToD1 = Import(FromD, Lang_CXX);
+  }
+
+  Decl *ToD2;
+  {
+    Decl *FromTU = getTuDecl(
+        R"(
+            struct A { int x; };
+            )",
+        Lang_CXX, "input1.cc");
+    auto *FromD = FirstDeclMatcher<CXXRecordDecl>().match(
+        FromTU, cxxRecordDecl(hasName("A")));
+
+    ToD2 = Import(FromD, Lang_CXX);
+  }
+  
+  EXPECT_EQ(ToD1, ToD2);
+}
+
+TEST_P(ASTImporterTestBase, ImportOfEquivalentRecordWithFunction) {
+  Decl *ToD1;
+  {
+    Decl *FromTU = getTuDecl(
+        R"(
+            struct A { int x(); };
+            )",
+        Lang_CXX, "input0.cc");
+    auto *FromD = FirstDeclMatcher<CXXRecordDecl>().match(
+        FromTU, cxxRecordDecl(hasName("A")));
+
+    ToD1 = Import(FromD, Lang_CXX);
+  }
+
+  Decl *ToD2;
+  {
+    Decl *FromTU = getTuDecl(
+        R"(
+            struct A { int x(); };
+            )",
+        Lang_CXX, "input1.cc");
+    auto *FromD = FirstDeclMatcher<CXXRecordDecl>().match(
+        FromTU, cxxRecordDecl(hasName("A")));
+
+    ToD2 = Import(FromD, Lang_CXX);
+  }
+  
+  EXPECT_EQ(ToD1, ToD2);
+}
+
+TEST_P(ASTImporterTestBase, ImportOfNonEquivalentRecordWithField) {
+  Decl *ToD1;
+  {
+    Decl *FromTU = getTuDecl(
+        R"(
+            struct A { int x; };
+            )",
+        Lang_CXX, "input0.cc");
+    auto *FromD = FirstDeclMatcher<CXXRecordDecl>().match(
+        FromTU, cxxRecordDecl(hasName("A")));
+
+    ToD1 = Import(FromD, Lang_CXX);
+  }
+
+  Decl *ToD2;
+  {
+    Decl *FromTU = getTuDecl(
+        R"(
+            struct A { unsigned x; };
+            )",
+        Lang_CXX, "input1.cc");
+    auto *FromD = FirstDeclMatcher<CXXRecordDecl>().match(
+        FromTU, cxxRecordDecl(hasName("A")));
+
+    ToD2 = Import(FromD, Lang_CXX);
+  }
+  
+  EXPECT_NE(ToD1, ToD2);
+}
+
+TEST_P(ASTImporterTestBase, DISABLED_ImportOfNonEquivalentRecordWithFunction) {
```

Comment:
I think it would worth to add an other similar test,  where we import the functionDecl(hasName('x')) and not the cxxRecordDecl itself. In that case structural eq will dispatch to comparing the methods, so `EXPECT_EQ(ToD1, ToD2)` will be successful.

Reply:
Now there is a ImportOf{Equivalent|NonEquivalent}RecordWith{Field|Function} test (4 tests). Is it OK to duplicate every of them with a version that imports the member `x` instead of the record `A`? (And there is the possibility to add the case with friends.) The import of `x` can be inserted into the existing tests too.

Reply:
> The import of x can be inserted into the existing tests too.

Perhaps that would be the best.